### PR TITLE
More advanced template

### PR DESCRIPTION
**Pull request by coffee-cup**
We want users to have everything they need when using create-prodo-app. Previously the prodo-template was very barebones. This update adds some features to the template.

- Route plugin and two routes (Home/About)
- Counter component on home page
- Update tests

_Note: the getting started guide will have to be updated to reflect these changes_

### CODE REVIEW

File: src/components/Counter/index.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+import "./styles.scss";
+
+const Counter = () => (
+  <div className="counter">
+    <button onClick={() => dispatch(actions.changeCount)(-1)}>-</button>
+    <h1 className="count" data-testid="counter">
+      {watch(state.count)}
+    </h1>
+    <button onClick={() => dispatch(actions.changeCount)(1)}>+</button>{" "}
```
Comment (andrejak) [inline src/components/Counter/index.tsx]: Is the space at the end intentional?

File: tests/App.test.tsx [hunk 2]
```diff
@@ -2,15 +2,24 @@ import { render } from "@testing-library/react";
...
+    const { getByTestId } = render(
+      <Provider>
+        <App />
+      </Provider>,
+    );
 
-  expect(container.textContent).toBe("Hello World");
+    expect(getByTestId("counter").textContent).toBe("0");
+  });
 });
```
Comment (andrejak) [inline tests/App.test.tsx]: Is it possible/worthwhile to have any tests for the routing?

File: src/components/Counter/index.tsx (hunk 1 above)
Comment (coffee-cup) [reply src/components/Counter/index.tsx]: nope :D

File: tests/App.test.tsx (hunk 2 above)
Comment (coffee-cup) [reply tests/App.test.tsx]: Probably worth it just so users can see how it is done.

**Comment by coffee-cup** [comment]
@andrejak after some discussion with @tdawes I've decided to remove the counter and about pages. The route plugin is still there, but there is only a single page. Hopefully this is a good starting point for users to read the tutorial and getting started guide.

**Comment by coffee-cup** [comment]
image

**Comment by coffee-cup** [comment]
image

**Comment by coffee-cup** [comment]
I've also added a favicon and robots.txt file. Something CRA already does.

**Pull request merged**